jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorManager.java

```diff
@@ -53,9 +53,18 @@ public MonitorManager(final Class<T> clazz, final MonitorDirectory<T> monitorDir
         this.monitorDirectory = monitorDirectory;
         this.constNamespaceSet = constNamespaceSet;
         if (id == null) {
-            id = Integer.toString(clazz.getClassLoader().hashCode());
+            id = getId(clazz);
         }
     }
+    /**
+     * Method to get monitor manager id derived from the class loader id of input class.
```

Comment:
Provides a unique id for MonitorManagers defined in class loader for web apps or jvm scope.

Reply:
Thanks. Done